.github/workflows/python_unit_test.yaml

```diff
@@ -28,22 +41,25 @@ jobs:
           --health-interval 10s
           --health-timeout 5s
           --health-retries 5
-    
+
     steps:
       - name: Check out code
         uses: actions/checkout@v3
+        with:
+          persist-credentials: false
       - name: Set up python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.6.7"
+          python-version: ${{ inputs.python-version }}
       - name: Install poetry
         run: |
-            pip --version
-            pip install poetry
-            poetry install -vv
+          pip install --upgrade pip
+          pip version
+          pip install poetry
+          poetry install -vv
       - name: Service specific setup
         run: |
-            curl -sSL "${{inputs.setup-script-location}}" | bash
+          curl -sSL "${{inputs.setup-script-location}}" | bash
```

Comment:
is this dangerous for community PRs / forks? It seems like perhaps an easy thing for a reviewer to miss if this is calling out to somewhere else. Can we restrict this in some way to require that the file it's calling be in the same git repo?

Reply:
Good point. We could standardize a location in a repo where that script would be stored. Like, it has to be called .setup_script or something. Then it just reads it automatically from the repo, and you don't have to fiddle around with managing setup script URLs, it's just right there in the repo you're testing, or it isn't.